websocketOutportDriver/sender/webSocketSender.go

```diff
@@ -74,37 +74,32 @@ func NewWebSocketSender(args WebSocketSenderArgs) (*webSocketSender, error) {
 	return ws, nil
 }
 
+// AddClient will add the client to internal maps and will also start
 func (w *webSocketSender) AddClient(wss WSConn, remoteAddr string) {
 	client := &webSocketClient{
 		conn:       wss,
 		remoteAddr: remoteAddr,
 	}
-	w.mutClients.Lock()
-	w.clients[remoteAddr] = client
-	w.mutClients.Unlock()
+
+	w.clientsHolder.AddClient(client)
 
 	w.mutAcknowledges.Lock()
-	w.acknowledges[remoteAddr] = make(map[uint64]struct{})
+	w.acknowledges[remoteAddr] = NewWebsocketClientAcknowledgesHolder()
 	w.mutAcknowledges.Unlock()
 
-	go w.handleReceiveAck(client)
-}
-
-func (w *webSocketSender) getClients() map[string]*webSocketClient {
-	w.mutClients.RLock()
-	defer w.mutClients.RUnlock()
+	if !w.withAcknowledge {
```

Comment:
this should be moved on L85 as we do not need a new entry in the acknowledges map